lens-cube/src/main/java/org/apache/lens/cube/metadata/CubeDimensionTable.java

```diff
@@ -46,8 +51,15 @@ public CubeDimensionTable(String dimName, String dimTblName, List<FieldSchema> c
 
   public CubeDimensionTable(String dimName, String dimTblName, List<FieldSchema> columns, double weight,
     Map<String, UpdatePeriod> snapshotDumpPeriods, Map<String, String> properties) {
+    this(dimName, dimTblName, columns, null, weight, snapshotDumpPeriods, properties);
+  }
+
+  public CubeDimensionTable(String dimName, String dimTblName, List<FieldSchema> columns, Set<String> partCols,
```

Comment:
Can we simply read partCols from property set with value as comma separated list? This will allow user to say those are the part columns. Because all these part columns need not be actually part columns in storage tables, its becomes basically property.

If it is not property and they are first class entity like this, we need to add support all the way from xsd, metastore service with current changes.

If we are going with property, please document property in cube xsd.